### PR TITLE
Add Okta integration

### PR DESCRIPTION
**Pull request by andrewstucki**
## What does this PR do?

This adds a package for the filebeat Okta module.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md#tips-for-building-integrations) and this pull request is aligned with them.
- [x] I have verified that all datasets collect metrics or logs.

## Screenshots

Screen Shot 2020-08-05 at 1 38 41 PM
Screen Shot 2020-08-05 at 1 38 59 PM
Screen Shot 2020-08-05 at 1 39 12 PM
Screen Shot 2020-08-05 at 1 39 53 PM
Screen Shot 2020-08-06 at 9 49 28 AM
Screen Shot 2020-08-05 at 1 42 42 PM
Screen Shot 2020-08-05 at 2 31 33 PM

### CODE REVIEW

File: packages/okta/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            required: false
+            show_user: false
+          - name: http_headers
+            type: text
+            title: HTTP Headers
+            multi: false
+            required: false
+            show_user: false
+          - name: http_request_body
+            type: text
```
Comment (andrewkroh) [inline packages/okta/manifest.yml:61]: Should this and `http_headers` be `yaml`? I think they are both objects/maps.

Comment (andrewstucki) [reply packages/okta/manifest.yml:61]: I'll try it out, also, looking for examples of enabling/disabling inputs by default, but didn't even realize that was possible.

Comment (andrewstucki) [reply packages/okta/manifest.yml]: So, looks like you just need to add an `enabled: false` to the input and voila:
Screen Shot 2020-08-06 at 2 22 42 PM

File: packages/okta/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+**Exported fields**
+
+| Field | Description | Type |
+|---|---|---|
+| @timestamp | Event timestamp. | date |
+| client.as.number | Unique number allocated to the autonomous system. | long |
+| client.as.organization.name | Organization name. | keyword |
+| client.domain | Client domain. | keyword |
+| client.geo.city_name | City name. | keyword |
```
Comment (leehinman) [inline packages/okta/docs/README.md:19]: looks like some of the geo fields are missing like geo.continent_name, etc.

Comment (andrewstucki) [reply packages/okta/docs/README.md:19]: Both these, and the `user_agent` fields are actually raw copied from the original okta event, so we don't do additional geoip/user_agent processing on the ingest pipeline as of now. I can update the ingest pipeline in a subsequent PR, but if we do, we should circle back and change the original filebeat module.

File: packages/okta/dataset/system/fields/ecs.yml [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+- description: Region name.
+  name: destination.geo.region_name
+  type: keyword
+- description: IP address of the destination.
+  name: destination.ip
+  type: ip
+- description: ECS version this event conforms to.
+  name: ecs.version
+  type: keyword
+- description: The action captured by the event.
```
[the remark this answers is not in the document]
Comment (andrewstucki) [reply packages/okta/dataset/system/fields/ecs.yml:70]: Thanks, looks like it is indeed missing.

File: packages/okta/manifest.yml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+            multi: false
+            required: false
+            show_user: true
+          - name: url
+            type: text
+            title: Okta System Log API Url
+            multi: false
+            required: false
+            show_user: true
+        title: "Collect Okta logs via webhook"
```
Comment (andrewkroh) [inline packages/okta/manifest.yml:118]: The term "webhook" makes me think that filebeat will be accepting webhook callbacks. Perhaps it should be "via API"?

**Pull request merged**